ksql-engine/src/test/java/io/confluent/ksql/function/udaf/topk/DoubleTopkKudafTest.java

```diff
@@ -21,77 +21,80 @@
 import org.junit.Before;
 import org.junit.Test;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.junit.Assert.assertThat;
 
 public class DoubleTopkKudafTest {
-  Object[] valueArray;
+  ArrayList valueArray;
   TopKAggregateFunctionFactory topKFactory;
   List<Schema> argumentType;
 
   @Before
   public void setup() {
-    valueArray = new Double[]{10.0, 30.0, 45.0, 10.0, 50.0, 60.0, 20.0, 60.0, 80.0, 35.0, 25.0};
+    valueArray = new ArrayList(Arrays.asList(10.0, 30.0, 45.0, 10.0, 50.0, 60.0, 20.0, 60.0, 80.0, 35.0,
+                                             25.0));
     topKFactory = new TopKAggregateFunctionFactory(3);
     argumentType = Collections.singletonList(Schema.FLOAT64_SCHEMA);
   }
 
   @Test
   public void shouldAggregateTopK() {
-    KsqlAggregateFunction<Object, Object[]> topkKudaf =
-            topKFactory.getProperAggregateFunction(argumentType);
-    Object[] window = new Double[]{null, null, null};
+    KsqlAggregateFunction<Object, ArrayList> topkKudaf =
+        topKFactory.getProperAggregateFunction(argumentType);
```

Comment:
Looks like we are missing some type information on `AggregateFunctionFactory` i.e., `getProperAggregateFunction` should return a `KsqlAggregateFunction<V, A>` then we get rid of all the warning noise in this class etc

Reply:
With the latest changes there is no warning.

Reply:
How would you add types to `AggregateFunctionFactory`?  It's subclasses can have multiple types for the given method.